pkg/odo/cli/component/push.go

```diff
@@ -82,15 +84,55 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
 
 	// if experimental mode is enabled and devfile is present
 	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(po.DevfilePath) {
-		envInfo, err := envinfo.NewEnvSpecificInfo(po.componentContext)
+		// We retrieve the configuration information. If this does not exist, then BLANK is returned (important!).
+		envFileInfo, err := envinfo.NewEnvSpecificInfo(po.componentContext)
 		if err != nil {
 			return errors.Wrap(err, "unable to retrieve configuration information")
 		}
-		po.EnvSpecificInfo = envInfo
+
+		// If the file does not exist, we should populate the environment file with the correct env.yaml information
+		// such as name and namespace.
+		// NOTE: Should we implement URL details later?
```

Comment:
Since we store URLs in the env.yaml in the .odo directory, this shouldn't be a problem as there is no .odo directory. Am I missing something?

Reply:
Ah, that's an old note, I'll remove it.